src/search/evaluators/sub_evaluator.cc

```diff
@@ -0,0 +1,98 @@
+#include "sub_evaluator.h"
+
+#include "../evaluation_context.h"
+#include "../evaluation_result.h"
+#include "../option_parser.h"
+#include "../plugin.h"
+
+#include <cstdlib>
+#include <sstream>
+#include <cmath>
+
+using namespace std;
+namespace sub_evaluator {
+    SubEvaluator::SubEvaluator(const Options &opts)
+            : evaluator(opts.get < shared_ptr < Evaluator >> ("eval")),
+              w(opts.get<double>("weight")),
+              t(opts.get<EvaluatorType>("type")) {
```

Comment:
I recommend using speaking variable names like "type" instead of "t". Using "w" for weight is fine, because its meaning is clear.